### PR TITLE
Update system requirements for 2.4.2

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this pull request

This pull request (PR) updates supported 3rd party software versions for the 2.4.2 release, plus some grammatical clean up I found along the way.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements.html

### CODE REVIEW

File: src/guides/v2.4/install-gde/system-requirements.md [hunk 1]
```diff
@@ -108,8 +110,8 @@ Mail Transfer Agent (MTA) or an SMTP server
 
 ## Technologies Magento can use
 
-*  [Redis][] version 5.0 is recommended and used in testing for page caching and session storage
-*  [Varnish]({{page.baseurl}}/config-guide/varnish/config-varnish.html) version 6.x (tested with 6.3.1)
+*  [Redis][] version 6.x is recommended for Magento 2.4.2. Magneto 2.4.x is compatible with Redis 5.
```
Comment (BarnyShergold) [inline src/guides/v2.4/install-gde/system-requirements.md:113]: Issue with using .x is you could argue that .x could be .3 .4 etc
```suggestion
*  [Redis][] version 6.x is recommended for Magento 2.4.2 and higher. Magento 2.4.1 and lower are compatible with Redis 5.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Once small typo and a suggestion - excellent clean up!

File: src/guides/v2.4/install-gde/system-requirements.md (hunk 1 above)
Comment (dobooth) [reply src/guides/v2.4/install-gde/system-requirements.md]: Good eye on the typo. I will leave the other as is. x does indeed subsume higher numbers, but the number of install errors on those versions is sure to be low. ;)

**Review by jeff-matthews — APPROVED** [review]

**Review by BarnyShergold — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Pull request merged**